Deprecate `containerPackageInfo` parameter in `createOdspCreateContainerRequest` (#23919)
microsoft/FluidFramework: .changeset/wicked-eagles-search.md

---
"@fluidframework/odsp-driver": minor
"__section": deprecation
---

> The `containerPackageInfo` parameter in `createOdspCreateContainerRequest()` is now deprecated.

Reviewer: ```suggestion
The containerPackageInfo parameter in createOdspCreateContainerRequest is now deprecated
```

Author: Applied changes

> The parameter `containerPackageInfo` in `createOdspCreateContainerRequest()` is deprecated and will be removed in version 2.40.

Reviewer 2: ```suggestion
Deprecate the `containerPackageInfo` parameter in `createOdspCreateContainerRequest()`
```

Author: Applied changes

Reviewer 3: @tylerbutler and @jzaffiro - I think Mario got comments both ways on whether the removal version should be included in this text.

> The `containerPackageInfo` parameter in `createOdspCreateContainerRequest()` is deprecated will be removed in version 2.40.0.

Reviewer: ```suggestion
The `containerPackageInfo` parameter in `createOdspCreateContainerRequest()` is deprecated and will be removed in version 2.40.0.
```

Author: Applied changes

> This will mean that the name of the containerPackage can no longer be sent through the request. Instead it can be added in the constructor of `OdspDriverUrlResolverForShareLink`.

Reviewer 2: ```suggestion
The name of the containerPackage can no longer be sent through the request. This functionality will be removed in version 2.40. Instead, it can be added in the constructor of `OdspDriverUrlResolverForShareLink`.
```

Author: Applied changes

> ---
> "@fluidframework/odsp-driver": minor
> ---
> ---
> "section": deprecation
> ---

Reviewer: ```suggestion
---
"@fluidframework/odsp-driver": minor
"__section": deprecation
---
```

The changeset format recently changed, but since this was created before the build-tools update, we can just fix it manually.

Author: Applied change

> Deprecate the `containerPackageInfo` parameter in `createOdspCreateContainerRequest()` is deprecated and will be removed in version 2.40.

Reviewer: ```suggestion
The containerPackageInfo parameter in createOdspCreateContainerRequest is now deprecated
```

Author: Applied change

> The name of the containerPackage can no longer be sent through the request. This functionality will be removed in version 2.40.

Reviewer: ```suggestion
The `containerPackageInfo` parameter in `createOdspCreateContainerRequest()` is deprecated will be removed in version 2.40.0.

The name of the containerPackage can no longer be sent through the request. Instead, it can be added in the constructor of `OdspDriverUrlResolverForShareLink`.
```

Author: Applied change

> Instead it can be added in the constructor of `OdspDriverUrlResolverForShareLink`.

Reviewer: ```suggestion
```

Author: Applied change

> See [issue](https://github.com/microsoft/FluidFramework/issues/23882) for details.

Reviewer: ```suggestion
See [issue #23822](https://github.com/microsoft/FluidFramework/issues/23882) for more details.
```

Author: Applied change

---
"@fluidframework/odsp-driver": minor
---
---
"section": deprecation
---

> createOdspCreateContainerRequest().containerPackageInfo is now deprecated

Reviewer 3: `createOdspCreateContainerRequest().containerPackageInfo` is not a thing.

Author: Applied change

> The parameter `containerPackageInfo` in `createOdspCreateContainerRequest()` is deprecated and will be removed in version 2.40.

Reviewer 3: Duplicate and somewhat at odds texts. While much the same one says "will be deprecated" and the other says "is deprecated". The second is correct. So just delete the first, I think.

Reviewer 3: Still duplicated - remove lines 8 and 9.
